### ISSUE TITLE
opam 2.0.0~rc2: git pins, git is truth

### ISSUE THREAD

**Issue by dbuenzli** [ocaml/opam#3408]
In a git pin it seems that now when I update the `opam` file in git it asks me whether I want to override the `opam` file. I don't really see why there's a reason for that:

This is basically an uninstalled `git`-pinned package in which I changed the `opam` file.

```
> opam upgrade omod 

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[omod.0.0.1] synchronised from git+file:///Users/dbuenzli/sync/repos/omod#master
[omod] Conflicting update of the metadata from git+file:///Users/dbuenzli/sync/repos/omod#master.
Override files in /Users/dbuenzli/.opam/4.03.0/.opam-switch/overlay/omod (there will be a backup)? 

omod is not installed. Install it? [Y/n] y
The following actions will be performed:
  - install omod 0.0.1*
```

**Comment by rjbou** [comment]
This should happen only opam file has been modified, _i.e._ the overlay opam is different from the old one & from the repository one. Are you in that case?

**Comment by dbuenzli** [comment]
No. It seems this happens all the time. I juste updated my `opam` file in `git` and from the above install state and it asks me again: 

```
> opam upgrade omod

<><> Synchronising pinned packages ><><><><><><><><><><><><><><><><><><><><><><>
[omod.0.0.1] synchronised from git+file:///Users/dbuenzli/sync/repos/omod#master
[omod] Conflicting update of the metadata from
git+file:///Users/dbuenzli/sync/repos/omod#master.
Override files in /Users/dbuenzli/.opam/4.03.0/.opam-switch/overlay/omod (there
will be a backup)? [Y/n] 
```

**Comment by rjbou** [comment]
Can you share overlay & versionned opam files?
Can you confirm your process:
- `git` pin `omod`, without installing it
- change & commit opam file
- upgrade `omod`

**Comment by rjbou** [comment]
Did you pinned `omod` with version setting :`opam pin omod.0.0.1 <dir>`?

**Comment by dbuenzli** [comment]
> Did you pinned omod with version setting :opam pin omod.0.0.1 <dir>?

No.

**Comment by rjbou** [comment]
So `omod` was already installed in version `0.0.1` before the local pin?

**Comment by dbuenzli** [comment]
It started as a new git-pinned package. The it got released, then it became installed via 0.0.1's distribution then it became pin again.

**Comment by dbuenzli** [comment]
(and meanwhile `opam` itself got upgraded from 2.0.0rc to 2.0.0rc~)

**Comment by rjbou** [comment]
> it became installed via 0.0.1's distribution then it became pin again.

Ok, the problem comes from here. If the package is installed, its version is on the switch state, so when it got pinned afterwards, it keep that version on overlay opam file. But when the comparison is made, it just retrieves version from the new opam file, and as the field is missing, it has `~dev` default ; which causes a conflict between the files.
